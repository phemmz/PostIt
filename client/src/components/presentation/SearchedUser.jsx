import React from 'react';
import PropTypes from 'prop-types';

/**
 * SearchedUser
 * @return {*} div
 */
const SearchedUser = ({ user }) => {
  return (
    <div className="card search-user center green-text text-darken-4">
      <span>Username: {user.username}</span>
      <br />
      <span>Email: {user.email}</span>
      <br />
      <span>Phone Number: {user.phoneNumber}</span>
    </div>
  );
};

SearchedUser.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string,
    email: PropTypes.string,
    phoneNumber: PropTypes.string
  }).isRequired
};

export default SearchedUser;
