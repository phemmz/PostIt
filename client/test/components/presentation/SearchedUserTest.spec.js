import React from 'react';
import { mount } from 'enzyme';
import SearchedUser from '../../../src/components/presentation/SearchedUser';

describe('SideNav Component', () => {
  const user = {
    username: 'phemmz',
    email: 'phemmzmcllroy@gmail.com',
    phoneNumber: '08012345678'
  };
  it('should render a sidenav component', () => {
    const wrapper = mount(<SearchedUser user={user} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('user')).toBe(user);
    expect(wrapper.find('div').exists()).toBe(true);
    expect(wrapper.find('span').exists()).toBe(true);
    expect(wrapper.find('br').exists()).toBe(true);
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.find('span').length).toBe(3);
    expect(wrapper.find('br').length).toBe(2);
  });
});
