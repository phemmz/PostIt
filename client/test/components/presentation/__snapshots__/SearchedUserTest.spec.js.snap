// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav Component should render a sidenav component 1`] = `
<SearchedUser
  user={
    Object {
      "email": "phemmzmcllroy@gmail.com",
      "phoneNumber": "08012345678",
      "username": "phemmz",
    }
  }
>
  <div
    className="card search-user center green-text text-darken-4"
  >
    <span>
      Username: 
      phemmz
    </span>
    <br />
    <span>
      Email: 
      phemmzmcllroy@gmail.com
    </span>
    <br />
    <span>
      Phone Number: 
      08012345678
    </span>
  </div>
</SearchedUser>
`;
